### PR TITLE
HW2

### PR DESCRIPTION
**Pull request by naztekinalp**
- Implement Two sum.
- Implement Length of Last Word.
- Implement Sqrt(x).
- Implement Fibonacci.
- Implement Fizz-buzz.

### CODE REVIEW

File: HW2/Fibonacci.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package HW2;
+public class Fibonacci {
+    public static void main(String args[]) {
+        int n = 100;
+        fibo(n);
```
Comment (drevalanya) [inline HW2/Fibonacci.java]: not working. try to run it in your idea

File: HW2/LengthOfLastWord.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package HW2;
+
+public class LengthOfLastWord {
+    public static void main(String[] args) {
+        lengthOfLastWord("hello test word");
```
Comment (drevalanya) [inline HW2/LengthOfLastWord.java]: incorrect result with input "   fly me   to   the moon  "

File: HW2/Fibonacci.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        while (count < n) {
+            System.out.println(number1 + " ");
+            int number3 = number2 + number1;
+            number1 = number2;
+            number2 = number3;
+            count = count++;
+        }
+    }
+
+}
```
Comment (drevalanya) [inline HW2/Fibonacci.java]: check style

File: HW2/Sqrt.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    public static void main(String[] args) {
+        mySqrt(4);
+    }
+
+    public static int mySqrt(int x) {
+        int result = 0;
+        if (x == 0) {
+            return 0;
+        } else {
+            result = (int) Math.sqrt(x);
```
Comment (drevalanya) [inline HW2/Sqrt.java]: Math.sqrt is built-in function 
image

File: HW2/Fibonacci.java (hunk 3 above)
Comment (naztekinalp) [reply HW2/Fibonacci.java]: style fix commited and pushed.

File: HW2/Sqrt.java (hunk 4 above)
Comment (naztekinalp) [reply HW2/Sqrt.java]: sqrt without built-in function commited and pushed.

File: HW2/Fibonacci.java (hunk 1 above)
Comment (naztekinalp) [reply HW2/Fibonacci.java]: Fibonacci.java updated, commited and pushed.

File: HW2/LengthOfLastWord.java (hunk 2 above)
Comment (naztekinalp) [reply HW2/LengthOfLastWord.java]: Trim string added to fix incorrect result -committed and pushed

File: HW2/Fibonacci.java (hunk 1 above)
Comment (drevalanya) [reply HW2/Fibonacci.java]: in this task you should print all numbers from Fibonacci line, not only the last one

Comment (naztekinalp) [reply HW2/Fibonacci.java]: change has been made for print all numbers

**Review by drevalanya — APPROVED** [review]
approved, but check style still not fixed

image

**Comment by naztekinalp** [comment]
Thanks for approving, added a style fix in the last commits.
[bd45e60](https://github.com/naztekinalpp/NazTekinalp/pull/5/commits/bd45e606c1b19b50fad8d0a6c969c3d1e5d76830)
[8d838de](https://github.com/naztekinalpp/NazTekinalp/pull/5/commits/8d838de9da45ed2277c707e4e1a0b857c63aee9b)
[fce73b0](https://github.com/naztekinalpp/NazTekinalp/pull/5/commits/fce73b0f0b351c823a6bf304b3d4a7b1e42fd454)

**Pull request merged**